src/scheduler.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_SCHEDULER_H
+#define BITCOIN_SCHEDULER_H
+
+//
+// NOTE:
+// boost::thread / boost::function / boost::chrono should be ported to
+// std::thread / std::function / std::chrono when we support C++11.
+//
+#include <boost/function.hpp>
+#include <boost/chrono/chrono.hpp>
+#include <boost/thread.hpp>
+#include <map>
+
+//
+// Simple class for background tasks that should be run
+// periodically or once "after a while"
+//
+// Usage:
+//
+// CScheduler s;
+// s.scheduleFromNow(doSomething, 11); // Assuming a: void doSomething() { }
+// s.scheduleFromNow(boost::bind(Class::func, this, argument), 3);
+// boost::thread* t = s.run();
+//
+// Thread lifetime management (interruption, joining, deletion) is left up to the caller.
+//
+class CScheduler
+{
+public:
+    CScheduler();
+
```

Comment:
Unsure about this, but leaving the thread running forever may cause destruction order issues if a task is running exactly at shutdown. A destructor that sets an 'exit' bool flag (which the scheduler thread checks), and waits for all scheduler threads to exit may be needed.